quantum/deutsch_jozsa.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+"""
+Deutsch-Josza Algorithm is one of the first examples of a quantum
+algorithm that is exponentially faster than any possible deterministic
+classical algorithm
+
+Premise:
+We are given a hidden Boolean function f, 
+which takes as input a string of bits, and returns either 0 or 1:
+
+f({x0,x1,x2,...}) -> 0 or 1, where xn is 0 or 1
+ 
+The property of the given Boolean function is that it is guaranteed to
+either be balanced or constant. A constant function returns all 0's 
+or all 1's for any input, while a balanced function returns  0's for 
+exactly half of all inputs and 1's for the other half. Our task is to
+determine whether the given function is balanced or constant.
+
+References:
+- https://en.wikipedia.org/wiki/Deutsch-Jozsa_algorithm
+- https://qiskit.org/textbook/ch-algorithms/deutsch-jozsa.html
+"""
+
+import numpy as np
+import qiskit as q
+
+
+def dj_oracle(case: str, n: int) -> q.QuantumCircuit:
```

Comment:
Perhaps this should be two separate functions.

Reply:
The variable name n is not self documenting.  Can we come up with a more descriptive name.

Reply:
changed var name from n to _num_qubits_

Reply:
However, I feel the function is fine as it is.
I think we shouldn't split the function.